### PR TITLE
[batch] batch attributes table

### PR DESCRIPTION
**Pull request by jigold**
Same PR as #5990, but I rebased it on current master and I'll debug it. FYI: @danking

### CODE REVIEW

File: batch/batch/batch.py [hunk 1]
```diff
@@ -759,6 +759,12 @@ def from_record(record, deleted=False):
             cancelled=False,
             closed=False)
 
+        if attributes is not None:
+            for k, v in attributes.items():
+                await db.batch_attributes.new_record(batch_id=id,
+                                                     key=k,
+                                                     value=v)
```
Comment (danking) [inline batch/batch/batch.py]: we should do all these in one sql query

File: batch/batch/database.py [hunk 2]
```diff
@@ -399,3 +451,31 @@ def __init__(self, db):
...
+    # async def new_records(self, items):
+    #     async with self._db.pool.acquire() as conn:
+    #         async with conn.cursor() as cursor:
+    #             if len(items) > 0:
+    #                 await cursor.executemany(self._batch_attributes_sql, items)
+    #                 n_inserted = cursor.rowcount
+    #                 if n_inserted != len(items):
+    #                     log.info(f'inserted {n_inserted} batch attributes, but expected {len(items)}')
+    #                     return False
+    #             return True
```
Comment (danking) [inline batch/batch/database.py]: delete

File: batch/batch/database.py [hunk 3]
```diff
@@ -399,3 +451,31 @@ def __init__(self, db):
 
     async def get_undeleted_records(self, ids, user):
         return await super().get_records({'id': ids, 'user': user, 'deleted': False})
+
+
+class BatchAttributesTable(Table):
+    # batch_attribute_fields = {'batch_id', 'key', 'value'}
```
Comment (danking) [inline batch/batch/database.py]: delete

File: batch/batch/database.py [hunk 4]
```diff
@@ -399,3 +451,31 @@ def __init__(self, db):
...
+
+    def __init__(self, db):
+        super().__init__(db, 'batch-attributes')
+        # self._batch_attributes_sql = self.new_record_template(BatchAttributesTable.batch_attribute_fields)
```
Comment (danking) [inline batch/batch/database.py]: delete

File: batch/create-batch-tables.sql [hunk 5]
```diff
@@ -55,6 +55,15 @@ CREATE TABLE IF NOT EXISTS `jobs-parents` (
...
 CREATE INDEX jobs_parents_parent_id ON `jobs-parents` (batch_id, parent_id);
 
+CREATE TABLE IF NOT EXISTS `batch-attributes` (
+  `batch_id` BIGINT NOT NULL,
+  `key` VARCHAR(100) NOT NULL,
+  `value` TEXT(65535),
+  PRIMARY KEY (`batch_id`, `key`),
+  FOREIGN KEY (`batch_id`) REFERENCES batch(id) ON DELETE CASCADE  
+) ENGINE = InnoDB;
+# CREATE INDEX batch_attributes_key_value ON `batch-attributes` (`key`(256), `value`(256));
```
Comment (danking) [inline batch/create-batch-tables.sql]: why the `#`?

**Comment by jigold** [comment]
See #6493

**Pull request merged**